src/Infrastructure/Decorators/DevAuthenticationService.cs

```diff
@@ -0,0 +1,98 @@
+    private readonly IAmazonCognitoIdentityProvider _provider;
+    private static readonly byte[] Key = Encoding.ASCII.GetBytes("This is a test key");
+
+    public DevAuthenticationService(IAuthenticationService authenticationService,
+        IOptionsMonitor<DevSettings> devSettings, IUserService userService, IHeaderService headerService,
+        AwsSettings awsSettings, IAmazonCognitoIdentityProvider provider)
+    {
+        _authenticationService = authenticationService;
+        _devSettings = devSettings;
+        _userService = userService;
+        _headerService = headerService;
+        _awsSettings = awsSettings;
+        _provider = provider;
+    }
+
+    private string CreateLocalIdToken(string id)
+    {
+        var tokenHandler = new JwtSecurityTokenHandler();
+        var tokenDescriptor = new SecurityTokenDescriptor
+        {
+            Subject = CreateClaimsIdentity(id),
+            Expires = DateTime.UtcNow.AddMinutes(10),
+            SigningCredentials = new SigningCredentials(new SymmetricSecurityKey(Key),
+                SecurityAlgorithms.HmacSha256Signature)
+        };
+        var token = tokenHandler.CreateToken(tokenDescriptor);
+        return tokenHandler.WriteToken(token);
+    }
+
+    private ClaimsIdentity CreateClaimsIdentity(string id)
+    {
+        return new ClaimsIdentity(new[]
+        {
+            new Claim("sub", id),
+            new Claim("email", "test@email.com"),
+            new Claim("email_verified", "true"),
+            new Claim("name", "John Doe"),
+            new Claim("iss", "https://cognito-idp.eu-west-2.amazonaws.com/us-west-1ABCDEFGHI"),
+            new Claim("cognito:username", id),
+            new Claim("cognito:groups", "Group1"),
+            new Claim("phone_number", "+447712345678"),
+            new Claim("phone_number_verified", "true")
+        });
+    }
+
+    public async Task<Response<SignUpResponse>> SignUpAsync(string email, string password, string selectedLocale)
+    {
+        var response = await _authenticationService.SignUpAsync(email, password, selectedLocale);
+        if (_devSettings.CurrentValue.AutoConfirmNewCognitoSignup)
+        {
+            await _provider.AdminConfirmSignUpAsync(new AdminConfirmSignUpRequest
+            {
+                Username = email,
+                UserPoolId = _awsSettings.CognitoPoolId,
+            });
+        }
+
+        return response;
+    }
+
+    public async Task<Response<string>> RespondToMfaChallengeAsync(string mfaCode, string mfaDetails)
+    {
+        if (_devSettings.CurrentValue.BypassMfa)
+        {
+            var mfaLoginDetails = _userService.DeserializeMfaLoginDetails(mfaDetails);
+            return Response<string>.CreateSuccessfulContentResponse(
+                CreateLocalIdToken(mfaLoginDetails.Username),
+                _headerService.GetConversationId());
+        }
+
+        return await _authenticationService.RespondToMfaChallengeAsync(mfaCode, mfaDetails);
```

Comment:
Put this in an else block because only one or other of these should happen.

This is distinct from `SignUpAsync()` where the inner service must be called in all cases.

Reply:
Surely the else block is redundant due to the return keyword?

Reply:
Functionally redundant yes, but not in terms of making the intent of the code explicit and more robust.

If someone changes the if block on line 88 so it doesn't return, then you fall through to call to the inner service which may be problematic. With an else block you will get a compiler error (which is a good thing).